perception/traffic_light_ssd_fine_detector/src/nodelet.cpp

```diff
@@ -146,10 +154,15 @@ void TrafficLightSSDFineDetectorNodelet::callback(
   const int batch_size = net_ptr_->getMaxBatchSize();
   while (num_rois != 0) {
     const int num_infer = (num_rois / batch_size > 0) ? batch_size : num_rois % batch_size;
-    auto data_d = cuda::make_unique<float[]>(num_infer * channel_ * width_ * height_);
-    auto scores_d = cuda::make_unique<float[]>(num_infer * detection_per_class_ * class_num_);
-    auto boxes_d = cuda::make_unique<float[]>(num_infer * detection_per_class_ * 4);
-    std::vector<void *> buffers = {data_d.get(), scores_d.get(), boxes_d.get()};
+    auto data_d = cuda::make_unique<float[]>(num_infer * input_shape_.size());
+    auto box_d = cuda::make_unique<float[]>(num_infer * box_dims_->size());
+    auto score_d = cuda::make_unique<float[]>(num_infer * score_dims_->size());
+    std::vector<void *> buffers;
+    if (dnn_header_type_.compare("pytorch") == 0) {
```

Comment:
I guess we can also detect binding order without this parameter by using `getBindingIndex` or newer TensorRT api.

Reply:
[dissucattion_memo]
To be updated after Autoware's TensorRT version is updated

Reply:
We will upgrade this block after the tensorrt version which autoware supports is upgraded.